File: swagger/java/src/main/java/io/swagger/client/model/V3SiriProductionTimetableSubscriptionRequest.java
/*
 * PTV Timetable API - Version 3
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * OpenAPI spec version: v3
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.client.model.V3SiriSubscriptionTopic;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import org.threeten.bp.OffsetDateTime;
/**
 * V3SiriProductionTimetableSubscriptionRequest
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2024-04-22T17:04:13.135709+10:00[Australia/Sydney]")

public class V3SiriProductionTimetableSubscriptionRequest {
  @SerializedName("start_time")
  private OffsetDateTime startTime = null;

  @SerializedName("end_time")
  private OffsetDateTime endTime = null;

  @SerializedName("subscriber_ref")
  private String subscriberRef = null;

  @SerializedName("subscription_ref")
  private String subscriptionRef = null;

  /**
   * Siri Message Format &#x27;xml&#x27; or &#x27;json&#x27;
   */
  @JsonAdapter(SiriFormatEnum.Adapter.class)
  public enum SiriFormatEnum {
    @SerializedName("0")
    NUMBER_0(0),
    @SerializedName("1")
    NUMBER_1(1);

    private Integer value;

    SiriFormatEnum(Integer value) {
      this.value = value;
    }
    public Integer getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }
    public static SiriFormatEnum fromValue(Integer input) {
      for (SiriFormatEnum b : SiriFormatEnum.values()) {
        if (b.value.equals(input)) {
          return b;
        }
      }
      return null;
    }
    public static class Adapter extends TypeAdapter<SiriFormatEnum> {
      @Override
      public void write(final JsonWriter jsonWriter, final SiriFormatEnum enumeration) throws IOException {
        jsonWriter.value(String.valueOf(enumeration.getValue()));
      }

      @Override
      public SiriFormatEnum read(final JsonReader jsonReader) throws IOException {
        Object value = jsonReader.nextInt();
        return SiriFormatEnum.fromValue((Integer)(value));
      }
    }
  }  @SerializedName("siri_format")
  private SiriFormatEnum siriFormat = null;

  @SerializedName("siri_version")
  private String siriVersion = null;

  @SerializedName("consumer_address")
  private String consumerAddress = null;

  @SerializedName("initial_termination_time")
  private OffsetDateTime initialTerminationTime = null;

  @SerializedName("topics")
  private List<V3SiriSubscriptionTopic> topics = new ArrayList<V3SiriSubscriptionTopic>();

  public V3SiriProductionTimetableSubscriptionRequest startTime(OffsetDateTime startTime) {
    this.startTime = startTime;
    return this;
  }

   /**
   * Siri Start Time of the Validity Period
   * @return startTime
  **/
  @Schema(required = true, description = "Siri Start Time of the Validity Period")
  public OffsetDateTime getStartTime() {
    return startTime;
  }

  public void setStartTime(OffsetDateTime startTime) {
    this.startTime = startTime;
  }

  public V3SiriProductionTimetableSubscriptionRequest endTime(OffsetDateTime endTime) {
    this.endTime = endTime;
    return this;
  }

   /**
   * Siri End Time of the Validity Period
   * @return endTime
  **/
  @Schema(required = true, description = "Siri End Time of the Validity Period")
  public OffsetDateTime getEndTime() {
    return endTime;
  }

  public void setEndTime(OffsetDateTime endTime) {
    this.endTime = endTime;
  }

  public V3SiriProductionTimetableSubscriptionRequest subscriberRef(String subscriberRef) {
    this.subscriberRef = subscriberRef;
    return this;
  }

   /**
   * Siri Subscriber Ref
   * @return subscriberRef
  **/
  @Schema(required = true, description = "Siri Subscriber Ref")
  public String getSubscriberRef() {
    return subscriberRef;
  }

  public void setSubscriberRef(String subscriberRef) {
    this.subscriberRef = subscriberRef;
  }

  public V3SiriProductionTimetableSubscriptionRequest subscriptionRef(String subscriptionRef) {
    this.subscriptionRef = subscriptionRef;
    return this;
  }

   /**
   * Siri Subscription Ref - Unique to a Subscriber Ref
   * @return subscriptionRef
  **/
  @Schema(required = true, description = "Siri Subscription Ref - Unique to a Subscriber Ref")
  public String getSubscriptionRef() {
    return subscriptionRef;
  }

  public void setSubscriptionRef(String subscriptionRef) {
    this.subscriptionRef = subscriptionRef;
  }

  public V3SiriProductionTimetableSubscriptionRequest siriFormat(SiriFormatEnum siriFormat) {
    this.siriFormat = siriFormat;
    return this;
  }

   /**
   * Siri Message Format &#x27;xml&#x27; or &#x27;json&#x27;
   * @return siriFormat
  **/
  @Schema(required = true, description = "Siri Message Format 'xml' or 'json'")
  public SiriFormatEnum getSiriFormat() {
    return siriFormat;
  }

  public void setSiriFormat(SiriFormatEnum siriFormat) {
    this.siriFormat = siriFormat;
  }

  public V3SiriProductionTimetableSubscriptionRequest siriVersion(String siriVersion) {
    this.siriVersion = siriVersion;
    return this;
  }

   /**
   * Siri Message Version &#x27;1.3&#x27; or &#x27;2.0&#x27;
   * @return siriVersion
  **/
  @Schema(required = true, description = "Siri Message Version '1.3' or '2.0'")
  public String getSiriVersion() {
    return siriVersion;
  }

  public void setSiriVersion(String siriVersion) {
    this.siriVersion = siriVersion;
  }

  public V3SiriProductionTimetableSubscriptionRequest consumerAddress(String consumerAddress) {
    this.consumerAddress = consumerAddress;
    return this;
  }

   /**
   * Siri Consumer Address - Baseline and Updates will be sent to this address
   * @return consumerAddress
  **/
  @Schema(required = true, description = "Siri Consumer Address - Baseline and Updates will be sent to this address")
  public String getConsumerAddress() {
    return consumerAddress;
  }

  public void setConsumerAddress(String consumerAddress) {
    this.consumerAddress = consumerAddress;
  }

  public V3SiriProductionTimetableSubscriptionRequest initialTerminationTime(OffsetDateTime initialTerminationTime) {
    this.initialTerminationTime = initialTerminationTime;
    return this;
  }

   /**
   * Siri Initial Termination Time - Expiry of the subscription
   * @return initialTerminationTime
  **/
  @Schema(required = true, description = "Siri Initial Termination Time - Expiry of the subscription")
  public OffsetDateTime getInitialTerminationTime() {
    return initialTerminationTime;
  }

  public void setInitialTerminationTime(OffsetDateTime initialTerminationTime) {
    this.initialTerminationTime = initialTerminationTime;
  }

  public V3SiriProductionTimetableSubscriptionRequest topics(List<V3SiriSubscriptionTopic> topics) {
    this.topics = topics;
    return this;
  }

  public V3SiriProductionTimetableSubscriptionRequest addTopicsItem(V3SiriSubscriptionTopic topicsItem) {
    this.topics.add(topicsItem);
    return this;
  }

   /**
   * Get topics
   * @return topics
  **/
  @Schema(required = true, description = "")
  public List<V3SiriSubscriptionTopic> getTopics() {
    return topics;
  }

  public void setTopics(List<V3SiriSubscriptionTopic> topics) {
    this.topics = topics;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3SiriProductionTimetableSubscriptionRequest v3SiriProductionTimetableSubscriptionRequest = (V3SiriProductionTimetableSubscriptionRequest) o;
    return Objects.equals(this.startTime, v3SiriProductionTimetableSubscriptionRequest.startTime) &&
        Objects.equals(this.endTime, v3SiriProductionTimetableSubscriptionRequest.endTime) &&
        Objects.equals(this.subscriberRef, v3SiriProductionTimetableSubscriptionRequest.subscriberRef) &&
        Objects.equals(this.subscriptionRef, v3SiriProductionTimetableSubscriptionRequest.subscriptionRef) &&
        Objects.equals(this.siriFormat, v3SiriProductionTimetableSubscriptionRequest.siriFormat) &&
        Objects.equals(this.siriVersion, v3SiriProductionTimetableSubscriptionRequest.siriVersion) &&
        Objects.equals(this.consumerAddress, v3SiriProductionTimetableSubscriptionRequest.consumerAddress) &&
        Objects.equals(this.initialTerminationTime, v3SiriProductionTimetableSubscriptionRequest.initialTerminationTime) &&
        Objects.equals(this.topics, v3SiriProductionTimetableSubscriptionRequest.topics);
  }

  @Override
  public int hashCode() {
    return Objects.hash(startTime, endTime, subscriberRef, subscriptionRef, siriFormat, siriVersion, consumerAddress, initialTerminationTime, topics);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3SiriProductionTimetableSubscriptionRequest {\n");
    
    sb.append("    startTime: ").append(toIndentedString(startTime)).append("\n");
    sb.append("    endTime: ").append(toIndentedString(endTime)).append("\n");
    sb.append("    subscriberRef: ").append(toIndentedString(subscriberRef)).append("\n");
    sb.append("    subscriptionRef: ").append(toIndentedString(subscriptionRef)).append("\n");
    sb.append("    siriFormat: ").append(toIndentedString(siriFormat)).append("\n");
    sb.append("    siriVersion: ").append(toIndentedString(siriVersion)).append("\n");
    sb.append("    consumerAddress: ").append(toIndentedString(consumerAddress)).append("\n");
    sb.append("    initialTerminationTime: ").append(toIndentedString(initialTerminationTime)).append("\n");
    sb.append("    topics: ").append(toIndentedString(topics)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: openapi.3.0.1/openapi/typescript-fetch/models/V3SiriProductionTimetableSubscriptionRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * PTV Timetable API - Version 3
 * The PTV Timetable API provides direct access to Public Transport Victoria\'s public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * The version of the OpenAPI document: v3
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { V3SiriSubscriptionTopic } from './V3SiriSubscriptionTopic';
import {
    V3SiriSubscriptionTopicFromJSON,
    V3SiriSubscriptionTopicFromJSONTyped,
    V3SiriSubscriptionTopicToJSON,
} from './V3SiriSubscriptionTopic';

/**
 * 
 * @export
 * @interface V3SiriProductionTimetableSubscriptionRequest
 */
export interface V3SiriProductionTimetableSubscriptionRequest {
    /**
     * Siri Start Time of the Validity Period
     * @type {Date}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    startTime: Date;
    /**
     * Siri End Time of the Validity Period
     * @type {Date}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    endTime: Date;
    /**
     * Siri Subscriber Ref
     * @type {string}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    subscriberRef: string;
    /**
     * Siri Subscription Ref - Unique to a Subscriber Ref
     * @type {string}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    subscriptionRef: string;
    /**
     * Siri Message Format 'xml' or 'json'
     * @type {number}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    siriFormat: V3SiriProductionTimetableSubscriptionRequestSiriFormatEnum;
    /**
     * Siri Message Version '1.3' or '2.0'
     * @type {string}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    siriVersion: string;
    /**
     * Siri Consumer Address - Baseline and Updates will be sent to this address
     * @type {string}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    consumerAddress: string;
    /**
     * Siri Initial Termination Time - Expiry of the subscription
     * @type {Date}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    initialTerminationTime: Date;
    /**
     * 
     * @type {Array<V3SiriSubscriptionTopic>}
     * @memberof V3SiriProductionTimetableSubscriptionRequest
     */
    topics: Array<V3SiriSubscriptionTopic>;
}


/**
 * @export
 */
export const V3SiriProductionTimetableSubscriptionRequestSiriFormatEnum = {
    NUMBER_0: 0,
    NUMBER_1: 1
} as const;
export type V3SiriProductionTimetableSubscriptionRequestSiriFormatEnum = typeof V3SiriProductionTimetableSubscriptionRequestSiriFormatEnum[keyof typeof V3SiriProductionTimetableSubscriptionRequestSiriFormatEnum];


/**
 * Check if a given object implements the V3SiriProductionTimetableSubscriptionRequest interface.
 */
export function instanceOfV3SiriProductionTimetableSubscriptionRequest(value: object): boolean {
    if (!('startTime' in value)) return false;
    if (!('endTime' in value)) return false;
    if (!('subscriberRef' in value)) return false;
    if (!('subscriptionRef' in value)) return false;
    if (!('siriFormat' in value)) return false;
    if (!('siriVersion' in value)) return false;
    if (!('consumerAddress' in value)) return false;
    if (!('initialTerminationTime' in value)) return false;
    if (!('topics' in value)) return false;
    return true;
}

export function V3SiriProductionTimetableSubscriptionRequestFromJSON(json: any): V3SiriProductionTimetableSubscriptionRequest {
    return V3SiriProductionTimetableSubscriptionRequestFromJSONTyped(json, false);
}

export function V3SiriProductionTimetableSubscriptionRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): V3SiriProductionTimetableSubscriptionRequest {
    if (json == null) {
        return json;
    }
    return {
        
        'startTime': (new Date(json['start_time'])),
        'endTime': (new Date(json['end_time'])),
        'subscriberRef': json['subscriber_ref'],
        'subscriptionRef': json['subscription_ref'],
        'siriFormat': json['siri_format'],
        'siriVersion': json['siri_version'],
        'consumerAddress': json['consumer_address'],
        'initialTerminationTime': (new Date(json['initial_termination_time'])),
        'topics': ((json['topics'] as Array<any>).map(V3SiriSubscriptionTopicFromJSON)),
    };
}

export function V3SiriProductionTimetableSubscriptionRequestToJSON(value?: V3SiriProductionTimetableSubscriptionRequest | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'start_time': ((value['startTime']).toISOString()),
        'end_time': ((value['endTime']).toISOString()),
        'subscriber_ref': value['subscriberRef'],
        'subscription_ref': value['subscriptionRef'],
        'siri_format': value['siriFormat'],
        'siri_version': value['siriVersion'],
        'consumer_address': value['consumerAddress'],
        'initial_termination_time': ((value['initialTerminationTime']).toISOString()),
        'topics': ((value['topics'] as Array<any>).map(V3SiriSubscriptionTopicToJSON)),
    };
}

